### ISSUE TITLE
Publish on npm with pre-built binary

### ISSUE THREAD

**Issue by nisargjhaveri** [ios-control/ios-deploy#509]
I know we actually recommend using brew for installing ios-deploy, but npm is also handy when working with js projects as we can specify this as a dependency directly without the need to externally install. I believe we're minting the npm package for the same. 

Currently, when installing the npm package, we build as a preinstall step.
Can we simply prebuild the binary and publish on npm to remove the build step at the install time?

This will save us some issues related to npm install we have, I believe. Which would make the npm package more reliable.

Thoughts? Suggestions?

**Comment by gabebear** [comment]
The majority of the issues I’ve seen that users had with npm come from npm supporting root-based installs or with our script to detect Xcode being present.

A lot of people used sudo with npm and added “unsafe-perm=true” to install ios-deploy. Once some npm packages are installed root globally then npm is a bit of a nightmare(everything installed global in npm also has to be root to work correctly).  Brew is better for regular command-line access mainly because it has saner/safer normal flows when installing as a command-line utility.

I think the script for detecting Xcode works better now… but brew is still better(Xcode is just a dependency you specify in brew).

**Comment by gabebear** [comment]
We have no plans to ever stop publishing to npm.

I’m not opposed to changing to a binary for npm if it would help people, I’m just not sure how much it would help. ios-deploy requires Xcode and compiles in seconds.

I also wouldn’t want to steer regular command-line users that way as npm installation issues from users were a pain.

**Comment by dnicolson** [comment]
Is it possible to publish version 1.12.0 to npm?

**Comment by gabebear** [comment]
We'll push the next release to npm.

With Xcode14-only features we didn't 100% vet issues compiling from source on a broad range of machines with the current 1.12.

Is there a feature you are needing?

**Comment by dnicolson** [comment]
Not really, I wanted to try out `--list_profiles` and installed using Homebrew for that.
